pubsub/cloud-client/subscriber.py

```diff
@@ -22,70 +22,91 @@
     """Lists all subscriptions for a given topic."""
-    pubsub_client = pubsub.Client()
-    topic = pubsub_client.topic(topic_name)
+    subscriber = pubsub_v1.SubscriberClient()
+    topic_path = subscriber.topic_path(project, topic_name)
 
-    for subscription in topic.list_subscriptions():
+    for subscription in subscriber.list_subscriptions(topic_path):
         print(subscription.name)
 
 
-def create_subscription(topic_name, subscription_name):
+def create_subscription(project, topic_name, subscription_name):
     """Create a new pull subscription on the given topic."""
-    pubsub_client = pubsub.Client()
-    topic = pubsub_client.topic(topic_name)
+    subscriber = pubsub_v1.SubscriberClient()
+    topic_path = subscriber.topic_path(project, topic_name)
+    subscription_path = subscriber.subscription_path(
+        project, subscription_name)
 
-    subscription = topic.subscription(subscription_name)
-    subscription.create()
+    subscription = subscriber.create_subscription(
+        subscription_path, topic_path)
 
-    print('Subscription {} created on topic {}.'.format(
-        subscription.name, topic.name))
+    print('Subscription created: {}'.format(subscription))
 
 
-def delete_subscription(topic_name, subscription_name):
+def delete_subscription(project, subscription_name):
     """Deletes an existing Pub/Sub topic."""
-    pubsub_client = pubsub.Client()
-    topic = pubsub_client.topic(topic_name)
-    subscription = topic.subscription(subscription_name)
+    subscriber = pubsub_v1.SubscriberClient()
+    subscription_path = subscriber.subscription_path(
+        project, subscription_name)
 
-    subscription.delete()
+    subscriber.delete_subscription(subscription_path)
 
-    print('Subscription {} deleted on topic {}.'.format(
-        subscription.name, topic.name))
+    print('Subscription deleted: {}'.format(subscription_path))
 
 
-def receive_message(topic_name, subscription_name):
-    """Receives a message from a pull subscription."""
-    pubsub_client = pubsub.Client()
-    topic = pubsub_client.topic(topic_name)
-    subscription = topic.subscription(subscription_name)
+def receive_messages(project, subscription_name):
+    """Receives messages from a pull subscription."""
+    subscriber = pubsub_v1.SubscriberClient()
+    subscription_path = subscriber.subscription_path(
+        project, subscription_name)
 
-    # Change return_immediately=False to block until messages are
-    # received.
-    results = subscription.pull(return_immediately=True)
+    def callback(message):
+        print('Received message: {}'.format(message))
+        message.ack()
 
-    print('Received {} messages.'.format(len(results)))
+    subscriber.subscribe(subscription_path, callback=callback)
 
-    for ack_id, message in results:
-        print('* {}: {}, {}'.format(
-            message.message_id, message.data, message.attributes))
+    # The subscriber is non-blocking, so we must keep the main thread from
```

Comment:
Is there any deliberately blocking method in the new API, similar to line 68 of the old sample?

Reply:
No, but see https://github.com/GoogleCloudPlatform/google-cloud-python/issues/3888